### PR TITLE
HomeHandler.get and post requires authentication

### LINKED ISSUE #24: Home page of remoteappmanager does not require authentication
Suppose a remoteappmanager is spawned for `user=username` and the reverse proxy is set for `/user/username`, anyone can access `https://hub_server_url/user/username` without logging in.

1. Log in as username
2. Open an incognito browser, go directly to https://hub_server_url/user/username

### PR DESCRIPTION
**Pull request by kitchoi**
Fix #24 

Should be revisited after #25

### CODE REVIEW

**Comment by kitchoi** [comment]
Tests pass, but handlers are not yet tested (issue #5 ) so there is a bit reduction in coverage...

**Comment by stefanoborini** [comment]
@kitchoi We need to address coverage in more detail as we proceed, but for now I think it's fine. We need to get a baseline and build from there.

**Comment by kitchoi** [comment]
@sbo Please review.

File: remoteappmanager/handlers/base_handler.py [hunk 1]
```diff
@@ -17,8 +28,35 @@ def render(self, template_name, **kwargs):
...
             user=self.current_user,
             base_url=self.application.config.base_url,
-            logout_url=urljoin(self.application.config.hub_prefix, "logout")
-        )
+            logout_url=urljoin(self.application.config.hub_prefix, "logout"))
 
         args.update(kwargs)
         super(BaseHandler, self).render(template_name, **args)
+
+    def verify_token(self, cookie_name, encrypted_cookie):
```
Comment (stefanoborini) [inline remoteappmanager/handlers/base_handler.py]: My only concern is that this code is not asyncronous, meaning that we assume the request operation to be quick. This code is from jupyterhub, and they do the same, but it's a potential concern for us.

I am going to merge this, but we might have to rework this routine with a threadpool (like in async docker client) and issue the request asynchronously.

**Comment by stefanoborini** [comment]
Not concerned about the codecov. We are going to aim for a baseline at 0.2 release, and move up from there.

**Comment by kitchoi** [comment]
I agree that we need to revisit the async issue.

**Pull request merged**